### PR TITLE
Add Mapping to Tenant on Tenant.ADD_ENTITY command

### LINKED ISSUE #23876: Add `Mapping` to `Tenant` on `Tenant.ADD_ENTITY` command
<h2>Description</h2>

- In the `TenantAddEntityProcessor` extend the `isEntityPresent` by checking the `MappingState` if a `Mapping` with the provided `entityKey` exists.
- Extend the `PersistedMapping` with an extra property:
    - `ArrrayProperty<StringValue> tenantIds`.
- In the `TenantEntityAddedApplier` add a case if the `entityType` == `MAPPING`. Inside this case add the tenant to the persisted mapping similar to how we already do for users.

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR adds tenantIds property into the PersistedMapping object and allow the TenantEntityAddedApplier to add the tenant id into the mapping state if the EntityType is MAPPING

## Related issues

closes #23876

### CODE REVIEW

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbMappingState.java [hunk 1]
```diff
@@ -78,6 +78,18 @@ public void addRole(final long mappingKey, final long roleKey) {
     }
   }
 
+  @Override
+  public void addTenant(final long mappingKey, final String tenantId) {
+    this.mappingKey.wrapLong(mappingKey);
+    final var fkClaim = claimByKeyColumnFamily.get(this.mappingKey);
+    if (fkClaim != null) {
+      final var fkKey = fkClaim.inner();
```
Comment (remcowesterhoud) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbMappingState.java:86]: 🔧 I would rename `fkKey` to `claim`. `fk` is used to refer to something that's a foreign key. Here you have a composite key of 2 strings.

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    tenantRecord.setEntityKey(entityKey).setEntityType(EntityType.MAPPING);
+
+    // when
+    tenantEntityAddedApplier.applyState(tenantKey, tenantRecord);
+
+    // then
+    assertThat(tenantState.getEntityType(tenantKey, entityKey).get().equals(EntityType.MAPPING));
+    final var persistedMapping = mappingState.get(entityKey).get();
+    assertThat(persistedMapping.getTenantIdsList()).containsExactly(tenantId);
+  }
```
Comment (remcowesterhoud) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java:61]: ❌ Could we add a same test case for users here? I know it's strictly not what this issue is about, but I suppose it was missed when user was implemented.

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/TenantEntityAddedApplier.java [hunk 3]
```diff
@@ -35,9 +35,9 @@ public void applyState(final long key, final TenantRecord tenant) {
       case MAPPING -> mappingState.addTenant(tenant.getEntityKey(), tenant.getTenantId());
       default ->
           throw new IllegalStateException(
-              "Unknown or unsupported entity type: '"
-                  + tenant.getEntityType()
-                  + "'. Please contact support for clarification.");
+              String.format(
+                  "Expected to add entity '%d' to tenant '%s', but entities of type '%s' cannot be added to tenants",
+                  tenant.getEntityKey(), tenant.getTenantId(), tenant.getEntityType()));
```
Comment (remcowesterhoud) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/TenantEntityAddedApplier.java:40]: 👍

**Review by remcowesterhoud — DISMISSED** [review]
Thanks @nicpuppa!

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java (hunk 2 above)
Comment (nicpuppa) [reply zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java:61]: Is the one above (?) 🤔

Comment (remcowesterhoud) [reply zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java:61]: There is only 1 test case in this class? What do you mean?

Comment (nicpuppa) [reply zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java:61]: I'll add a test case

Comment (remcowesterhoud) [reply zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/TenantAppliersTest.java:61]: I thought I was going crazy 😄 Thank you!

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantAddEntityProcessor.java [hunk 4]
```diff
@@ -79,7 +82,7 @@ final var record = command.getValue();
     }
 
     final var entityKey = record.getEntityKey();
-    if (!isEntityPresent(entityKey, record)) {
+    if (!isEntityPresent(entityKey, record.getEntityType())) {
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantAddEntityProcessor.java:85]: And reject the command if already assigned ?

[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantAddEntityProcessor.java:85]: Yep

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/PersistedMapping.java [hunk 5]
```diff
@@ -77,4 +81,22 @@ public PersistedMapping addRoleKey(final long roleKey) {
     roleKeysProp.add().setValue(roleKey);
     return this;
   }
+
+  public List<String> getTenantIdsList() {
```
Comment (koevskinikola) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/PersistedMapping.java:85]: 🙃 We can only name this `getTenantIds()` as it makes it more consistent with the setter.

**Review by koevskinikola — APPROVED** [review]
LGTM now @nicpuppa, just a small nitpick below 👍

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/PersistedMapping.java (hunk 5 above)
Comment (nicpuppa) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/PersistedMapping.java:85]: I prefer to use this naming convention, it's consistent with other Persisted record (eg. PersistedUser). I'll rather change the set name 👍

**Pull request merged**